### PR TITLE
Test the opam version

### PR DESCRIPTION
**Pull request by samoht**
Just a proof of concept, not tested at all so might not be working as expected.

### CODE REVIEW

**Comment by samoht** [comment]
I've updated the patch to add `--bypass-opam-version-check` as discussed on one the last mirage calls.

**Comment by samoht** [comment]
Note: this patch will allow us to drop support for opam 1.1 in mirage.

**Comment by samoht** [comment]
Pushed a new version with a simpler command-line name.

File: lib/mirage.ml [hunk 1]
```diff
@@ -2332,12 +2332,23 @@ let configure_makefile t =
...
   | [] -> ()
   | ps ->
-    if command_exists "opam" then opam "install" ps
+    if command_exists "opam" then
+      if !no_opam_version_check_ then ()
+      else (
+        let opam_version = read_command "opam --version" in
+        match split opam_version '.' with
+        | "1"::"2"::_ -> opam "install" ps
+        | _ -> error "Your version of opam: %s is not up-to-date. \
```
Comment (talex5) [inline lib/mirage.ml]: Err... isn't this also going to reject opam 1.3 and all later versions?

Comment (samoht) [inline lib/mirage.ml]: indeed. Not vey urgent because opam 1.3 is not ready yet but I'll fix that

**Pull request merged**